ml_tools/trackdatabase.py

```diff
@@ -10,40 +10,33 @@
 import h5py
 import os
 import logging
+import filelock
 from multiprocessing import Lock
 import numpy as np
 
-# default lock for safe database writes.
-#
-# note for multiprocessing this will need to be overwritten with a shared lock for each process.
-# which can be done via
-#
-# def init_workers(lock):
-#    trackdatabase.HDF5_LOCK = lock
-#
-# pool = multiprocessing.Pool(self.workers_threads, initializer=init_workers, initargs=(shared_lock,))
-
-HDF5_LOCK = Lock()
-
 
 class HDF5Manager:
     """ Class to handle locking of HDF5 files. """
 
+    LOCK_FILE = "/var/lock/classifier-hdf5.lock"
+
     def __init__(self, db, mode="r"):
         self.mode = mode
         self.f = None
         self.db = db
+        self.lock = filelock.FileLock(HDF5Manager.LOCK_FILE, timeout=10)
```

Comment:
This timeout seems a little small if the machine is bogged down and something is doing a big write. Something like 3 mins sounds safer to me